### PR TITLE
fix(upgrade): propagate return value of resumeBootstrap

### LINKED ISSUE #22723: upgrade patch of resumeBootstrap removes the return value
## I'm submitting a...
<pre><code>[X] Bug report</code></pre>

## Current behavior
`upgrade/static` [overrides resumeBootstrap](https://github.com/angular/angular/blob/5.2.8/packages/upgrade/src/static/upgrade_module.ts#L271) but does not return anything.

Protractor [reads this return value](https://github.com/angular/protractor/blob/02746dd37726b2647f8f8dbeb1152cd7f902acb0/lib/browser.ts#L967-L968) and potentially uses it later to [retrieve the $injector](https://github.com/angular/protractor/blob/5.3.0/lib/clientsidescripts.js#L103-L110).

## Expected behavior
Overridden methods should not change the method signature.

### PR DESCRIPTION
**Pull request by jbedard**
Fixes #22723 

I wonder of the same should be done for [the bootstrap call](https://github.com/jbedard/angular/blob/8aa5d194ecc882c691179c6635091e4e37d1ab08/packages/upgrade/src/static/upgrade_module.ts#L262)? That isn't quite the same since it isn't monkey-patching it, however `upgrade.bootstrap()` does replace `angular.bootstrap()` so consistent signatures would be nice. The dynamic version already [has a return value](https://github.com/jbedard/angular/blob/8aa5d194ecc882c691179c6635091e4e37d1ab08/packages/upgrade/src/dynamic/upgrade_adapter.ts#L406), but maybe that inconsistency wouldn't matter since it is deprecated.

### CODE REVIEW

File: packages/upgrade/test/dynamic/upgrade_spec.ts [hunk 1]
```diff
@@ -2860,6 +2860,28 @@ withEachNg1Version(() => {
...
+
+          adapter.bootstrap(element, ['ng1']).ready((ref) => {
+            ref.dispose();
+          });
+
+          tick(100);
+
+          const value = (<any>window).angular.resumeBootstrap();
+          expect(value).toBeDefined();
+          expect(value.invoke).toBeDefined();
```
Comment (gkalpak) [inline packages/upgrade/test/dynamic/upgrade_spec.ts]: Could you also assert that this is the same as `ref.ng1Injector` (or `angular.element(element).injector()`)?

File: packages/upgrade/test/static/integration/testability_spec.ts [hunk 2]
```diff
@@ -48,6 +48,22 @@ withEachNg1Version(() => {
...
+          const element = html('<div></div>');
+          window.name = 'NG_DEFER_BOOTSTRAP!' + window.name;
+
+          bootstrap(platformBrowserDynamic(), Ng2Module, element, ng1Module);
+
+          tick(100);
+
+          const value = (<any>window).angular.resumeBootstrap();
+          expect(value).toBeDefined();
+          expect(value.invoke).toBeDefined();
```
Comment (gkalpak) [inline packages/upgrade/test/static/integration/testability_spec.ts]: Could you also assert that this is the same as `upgrade.$injector` (or `angular.element(element).injector()`)?
(The `bootstrap` helper returns a promise that is resolved to `upgrade`.)

**Review by gkalpak — APPROVED** [review]

**Comment by gkalpak** [comment]
Also, linting fails 😱 (Running `gulp format` should fix it.)

**Comment by jbedard** [comment]
I think this is ready...

File: packages/upgrade/test/static/integration/testability_spec.ts [hunk 3]
```diff
@@ -48,6 +48,25 @@ withEachNg1Version(() => {
...
+         window.name = 'NG_DEFER_BOOTSTRAP!' + window.name;
+
+         bootstrap(platformBrowserDynamic(), Ng2Module, element, ng1Module);
+
+         tick(100);
+
+         const value = (<any>window).angular.resumeBootstrap();
+         expect(value).toBe(a1Injector);
+
+         tick(100);
```
Comment (gkalpak) [inline packages/upgrade/test/static/integration/testability_spec.ts]: Is this to flush remaining pending tasks (and stop `fakeAsync()` from complaining)?
Would [flush()][1] also work (which is more explicit imo)?

[1]: https://github.com/jbedard/angular/blob/6b7bf7ed07a6f0fa18f6175b8f982dc05409c442/packages/core/testing/src/fake_async.ts#L130

**Review by gkalpak — APPROVED** [review]

File: packages/upgrade/test/static/integration/testability_spec.ts (hunk 3 above)
Comment (jbedard) [reply packages/upgrade/test/static/integration/testability_spec.ts]: I agree. Updated.

**Comment by IgorMinar** [comment]
http://test/OCL:189843546:BASE:189843569:1521593320895:38c4225d

**Comment by IgorMinar** [comment]
https://test.corp.google.com/ui#id=OCL:184784396:BASE:189843746:1521593426800:63ed54f5

**Pull request closed without merging**